src/main/java/com/google/devtools/build/lib/analysis/IncompatiblePlatformProvider.java

```diff
@@ -69,7 +70,16 @@ public static IncompatiblePlatformProvider incompatibleDueToConstraints(
       @Nullable Label targetPlatform, ImmutableList<ConstraintValueInfo> constraints) {
     Preconditions.checkNotNull(constraints);
     Preconditions.checkArgument(!constraints.isEmpty());
-    return new AutoValue_IncompatiblePlatformProvider(targetPlatform, null, constraints);
+
+    // Deduplicate and sort the list of constraints. Doing it here means that everyone inspecting
+    // this provider doesn't have to deal with it.
+    ImmutableSet<ConstraintValueInfo> filteredConstraints = ImmutableSet.copyOf(constraints);
```

Comment:
It'd be easier to do this in one step (and to re-use the existing name, so that the old data can't be referenced):

```
constraints = constreams.stream()
    .sorted(Comparator.comparing(ConstraintValueInfo::label))
    .distinct()
    .collect(toImmutableList());

Reply:
Done